modules/apps/data-engine/data-engine-taglib/src/main/resources/META-INF/resources/data_layout_builder/js/utils/saveDataDefinition.es.js

```diff
@@ -72,63 +75,110 @@ const normalizeField = (availableLanguageIds, field) => {
 		nestedDataDefinitionFields: field.nestedDataDefinitionFields.length
 			? field.nestedDataDefinitionFields.map((nestedField) => ({
 					...nestedField,
-					...normalizeField(availableLanguageIds, nestedField),
+					...normalizeField(
+						availableLanguageIds,
+						nestedField,
+						defaultLanguageId
+					),
 			  }))
 			: [],
 	};
 };
 
+/**
+ * Normalize DD and DL state
+ * @param {object} dataDefinition
+ * @param {object} dataLayout
+ * @param {String?} defaultLanguageId
+ * @returns {object} {normalizedDataLayout, normalizedDataDefinition}
+ * @description This Normalization is necessary to handle
+ * differences data structures between across the components, such as
+ * App Builder, Data Engine and Forms
+ */
+
+export const normalizeState = (
```

Comment:
Keven, sorry about that, I know we already talked about it but I think we can separate this function in other functions like `normalizedDataLayout` and `normalizedDataDefinition` because, I saw in some other places in the project, we call these functions separated. Do you agree with it?